servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1783,6 +1806,166 @@ private void setStateFromEntry(SetStateEntry entry) throws FileDoesNotExistExcep
   }
 
   /**
+   * Sets the acl of a file or directory.
+   *
+   * @param path to be set acl on
+   * @param options acl option to be set
+   * @return true if set successfully, false otherwise
+   * @throws TachyonException
+   */
+  public boolean setAcl(TachyonURI path, SetAclOptions options) throws TachyonException {
+    if (options.getOwner() != null) {
+      return setOwner(path, options.getOwner(), options.isRecursive());
```

Comment:
In fact, there's a union type in Thrift https://thrift.apache.org/docs/idl#union. Maybe you can use it.